lua/lsp-format/init.lua

```diff
@@ -131,25 +131,30 @@ M.on_attach = function(client)
         M.buffers[bufnr] = {}
     end
     table.insert(M.buffers[bufnr], client.id)
-    local format_options = M.format_options[vim.bo.filetype] or {}
 
-    local event = "BufWritePost"
-    if format_options.sync then
-        event = "BufWritePre"
-    end
-
-    local group = vim.api.nvim_create_augroup("Format", { clear = false })
-
-    vim.api.nvim_clear_autocmds {
-        buffer = bufnr,
-        group = group,
-    }
-    vim.api.nvim_create_autocmd(event, {
-        group = group,
-        desc = "format on save",
-        pattern = "<buffer>",
-        callback = M.format,
+    local format_options = vim.tbl_deep_extend('keep', M.format_options[vim.bo.filetype] or {}, {
+      on_save = true
     })
+
+    if format_options.on_save then
```

Comment:
instead of wrapping everything in this if block, it would be cleaner to return early if `on_save` is false.

```lua
if not format_options.on_save then
    return
end
```

Reply:
`on_save` should only affect this block. If return early, codes for other features cannot put after the "return early" line in future. So I think wrapping in block is better.

Reply:
if the code changes in the future, this can be refactored.
Big nested if blocks are not nice, please change it